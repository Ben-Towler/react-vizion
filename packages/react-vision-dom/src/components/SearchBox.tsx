import React, { Component } from 'react';

type Props = {
  currentRefinement?: string;
  searchAsYouType?: boolean;
  className?: string;
  style?: React.CSSProperties;

  autoFocus?: boolean;

  onSearch?: (event: any) => void;
  onClear?: (event: any) => void;
  onFocus?: (event: React.FocusEvent) => void;
  onBlur?: (event: React.FocusEvent) => void;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
  onKeyPress?: (event: React.KeyboardEvent<HTMLInputElement>) => void;

  clearButton?: React.ReactNode;
  submitButton?: React.ReactNode;
  loadingIndicator?: React.ReactNode;
};

type State = {
  query: string;
};

class SearchBox extends Component<Props, State> {
  state = {
    query: '',
  };

  onSubmit = () => {
    const { query } = this.state;

    console.log(query);
  };

  onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const query = event.target.value;
    const { searchAsYouType } = this.props;

    this.setState({ query });

    if (searchAsYouType) {
      this.onSubmit();
    }
  };

  render() {
    const {
      className,
      onSubmit,
      onChange,
      onKeyPress,
      onBlur,
      autoFocus,
    } = this.props;

    return (
      <div className={className}>
        <form onSubmit={onSubmit ? onSubmit : this.onSubmit}>
          <input
            onChange={onChange ? onChange : this.onChange}
            onKeyPress={onKeyPress}
            onBlur={onBlur}
            autoFocus={autoFocus}
          />
          <button type="submit" />
        </form>
      </div>
    );
  }
}

export default SearchBox;
