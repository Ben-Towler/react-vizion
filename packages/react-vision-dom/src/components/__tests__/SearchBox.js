import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SearchBox from '../SearchBox';

Enzyme.configure({ adapter: new Adapter() });

describe('SearchBox', () => {
  it('applies its default props', () => {
    const instance = renderer.create(<SearchBox refine={() => null} />);

    expect(instance.toJSON()).toMatchSnapshot();

    instance.unmount();
  });

  describe('Event tests', () => {
    it('should change state on default onChange', () => {
      const component = shallow(<SearchBox />);
      const onChangeValue = 'test';

      component
        .find('input')
        .simulate('change', { target: { value: onChangeValue } });

      expect(component.state().query).toEqual(onChangeValue);
    });
  });
});
