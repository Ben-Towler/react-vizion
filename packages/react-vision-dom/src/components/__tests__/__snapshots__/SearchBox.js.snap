// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox applies its default props 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
    />
    <button
      type="submit"
    />
  </form>
</div>
`;
